home/home9.1/src/main/java/com/example/MyDataType.java

```diff
@@ -30,24 +25,38 @@ public Object getData() {
     }
 
     public static void serialize(MyDataType obj, OutputStream out) throws IOException {
-        MessageBufferPacker packer = MessagePack.newDefaultBufferPacker();
-        serializeData(obj.getData(), packer);
-        packer.close();
-        out.write(packer.toByteArray());
+        serializeData(obj.getData(), out);
     }
 
-    private static void serializeData(Object data, MessageBufferPacker packer) throws IOException {
+    private static void serializeData(Object data, OutputStream out) throws IOException {
         if (data instanceof Integer) {
-            packer.packInt((int) data);
+            serializeInt((int) data, out);
         } else if (data instanceof String) {
-            packer.packString((String) data);
+            serializeString((String) data, out);
         } else if (data instanceof List) {
-            List<Map.Entry<String, MyDataType>> list = (List<Map.Entry<String, MyDataType>>) data;
-            packer.packMapHeader(list.size());
-            for (Map.Entry<String, MyDataType> entry : list) {
-                packer.packString(entry.getKey());
-                serializeData(entry.getValue().getData(), packer);
-            }
+            serializeMap((List<Map.Entry<String, MyDataType>>) data, out);
+        }
+    }
+
+    private static void serializeInt(int value, OutputStream out) throws IOException {
```

Comment:
it can work that way for both string and int but to make effective serialisation there is a special way to serialise short strings and small integers: please check https://github.com/msgpack/msgpack/blob/master/spec.md#str-format-family for fixstr and fixint